### PR TITLE
Bump libnetwork for 1.7 release.

### LINKED ISSUE #13426: libnetwork: different behavior while creating bridge with IPv6 support compared to v1.5 and v1.6
As described in the [snapshot documentation](http://docs.master.dockerproject.com/articles/networking/#ipv6-with-docker) Docker versions 1.5 and 1.6 did setup a route and enabled IPv6 routing to Docker's fixed-cidr-v6 subnet (e.g. `2001:db8:1::/64`) while creating the docker0 bridge:

    $ ip -6 route add 2001:db8:1::/64 dev docker0
    $ sysctl net.ipv6.conf.default.forwarding=1
    $ sysctl net.ipv6.conf.all.forwarding=1

Currently, with libnetwork merged in 1.7.0-dev, these steps are not executed.

Should those steps be executed manually by the user or automatically when starting the Docker daemon?

Either we have to adjust the docs or we have to add functionality.

### PR DESCRIPTION
**Pull request by calavera**
Fixes #13426

[Changes between both commits](https://github.com/docker/libnetwork/compare/2da2dc055de5a474c8540871ad88a48213b0994f...4ded6fe3641b71863cc5985652930ce40efc3af4)

/cc @mavenugo, @mrjana 

Signed-off-by: David Calavera david.calavera@gmail.com

### CODE REVIEW

**Comment by icecrime** [comment]
@calavera You also have to run the `vendor.sh` script to actually update the content.

**Comment by calavera** [comment]
@icecrime done

**Comment by icecrime** [comment]
LGTM, waiting on approval from @mavenugo @mrjana or @aboch!

**Comment by mrjana** [comment]
LGTM

**Comment by icecrime** [comment]
Waiting on CI, thanks!

**Comment by mavenugo** [comment]
@icecrime @calavera should we have this PR pulled against @jfrazelle bump branch directly instead of master. We have an upcoming PR to docker/master from libnetwork/master which contains bulkier changes. considering we have more upcoming RCs, it would be better to have these cherry-picked changes from 1.7.0_integ branch in libnetwork into bump branch. WDYT ?

**Comment by icecrime** [comment]
@mavenugo We always merge to master and cherry-pick to the release branch, so that's the proper way to go. What do you mean by "we have more upcoming RC"? So far we plan to having an rc2 in the upcoming days, but it might very well be the last.

**Comment by calavera** [comment]
@icecrime I don't think the changes that @mavenugo mentions are for 1.7. There have been new additions in libnetwork that are not necessary for 1.7.

@mavenugo the process is to merge to master first. I picked the commit based on the last commit on `1.7.0_integ`. I think it's better to follow the stablished process, merge this PR first to move forward with the release and then open a new PR with the new changes.

**Comment by mavenugo** [comment]
@icecrime okay. we (libnetwork) vendor-in must be done in sequence in order to balance the features from master vs fixes for 1.7.0. Considering the fact that we dont have a foreseeable fixes targeting just the 1.7 release, am fine with this PR. 
We can deal with any upcoming sequencing issues down the line if we face it.

LGTM.

**Comment by estesp** [comment]
LGTM

**Comment by estesp** [comment]
MOAR cherry-picking for you @jfrazelle :)

File: vendor/src/github.com/docker/libnetwork/network.go [hunk 1]
```diff
@@ -60,14 +60,23 @@ type network struct {
 }
 
 func (n *network) Name() string {
+	n.Lock()
+	defer n.Unlock()
+
 	return n.name
```
Comment (tiborvass) [inline vendor/src/github.com/docker/libnetwork/network.go]: When guarding one access, let's avoid defer:
```Go
n.Lock()
name := n.name
n.Unlock()
return name
```

**Comment by jfrazelle** [comment]
cherry-picked

File: vendor/src/github.com/docker/libnetwork/network.go (hunk 1 above)
Comment (mrjana) [inline vendor/src/github.com/docker/libnetwork/network.go]: @tiborvass In this particular case I don't necessarily agree that we should avoid defer. We dont need extra boiler plate of introducing a temporary variable just for this. defer works nicely here and helps readabiity.

**Pull request merged**